### DISCUSSION TITLE
How to handle action you are already logged in with a different user

### DISCUSSION THREAD

**Discussion by dglozano** [keycloak/keycloak#32456] [Q&A]
### Description

If a user clicks on an action link that he or she received on an email but it's currently logged in with a different user, an error like below will be shown:

<img width="1143" alt="bilde" src="https://github.com/user-attachments/assets/03ac8a1f-42ed-4e98-bca6-c2bb86286105">

In such scenario, I would like to be able to make it easier to logout and retry the action.
Therefore, I was thinking I could customize the error.ftl template to show a link to the OIDC logout endpoint.

When going to OIDC logout endpoint, no information is shown about the currently logged in user. 

<img width="1146" alt="bilde" src="https://github.com/user-attachments/assets/284508b8-c637-49d3-abac-554d188348c3">

I would like to show who is the currently logged in user before clicking "logout" button (or even don't show the button if no session found?)

So my questions are:

 1. Is there any way to automatically logout the user instead of showing the error when verifying an email/clicking on update password?
 2. If not, is it possible to show a link that on click will logout the user and then immediately try to execute the action again?
 3. And kinf of unrelated, is it possible to show the information of the current logged in user when customising the freemarker templates? Like for example, in the logout template

**Comment by Saad-Shaikh** [comment]
Bumping this as I have the same issue. I wanted to add a link to the phrase 'sign out' in this template which the user can click and be signed out. After looking around a bit, I found out that it might be possible to get the logout url from the client object in error.ftl as `client.frontChannelLogoutUrl`. 
However, the client is not always present _(for example when the required action was triggered by an admin from the KC console)_. 

Is there is better way to do this which ensures the link is always present?
